### ISSUE TITLE
No validationError raised on multiple "SET" operators in UpdateExpression

### ISSUE THREAD

**Issue by nirmachlev** [getmoto/moto#7726]
Hi,
I have encountered the following discrepancy between production dynamodb and the moto mock:

Running a `transact_write_items` request with two "SET" operators under UpdateExpression causes `ValidationException` on dynamodb, while my mock db accepts it and my tests pass.

Example query:
```
    self._client.transact_write_items(
        TransactItems=[
            {
                "Update": {
                    "TableName": self._table_name,
                    "Key": {
                        "part_id": {"N": str(condition_state.part_id)},
                    },
                    "ConditionExpression": "version = :version AND is_final = :false",
                    "UpdateExpression": "SET expire_at_utc = :expire_at_utc SET is_final = :true ADD version :one",
                    "ExpressionAttributeValues": {
                        ":version": {"N": str(condition_state.version)},
                        ":true": {"BOOL": True},
                        ":false": {"BOOL": False},
                        ":one": {"N": "1"},
                        ":expire_at_utc": {"N": str(int(state_item.expire_at_utc.timestamp()))},
                    },
                }
            },
            {"Put": {"TableName": self._table_name, "Item": dynamo_item, "ConditionExpression": "attribute_not_exists(version)"}},
        ]
    )
```

Dynamodb raises the following exception:

```
File "/opt/python/botocore/client.py", line 1021, in _make_api_call
    raise error_class(parsed_response, operation_name)
botocore.exceptions.ClientError: An error occurred (ValidationException) when calling the TransactWriteItems operation: Invalid UpdateExpression: The "SET" section can only be used once in an update expression;

```
While moto doesn't.

**Comment by bblommers** [comment]
Hi @nirmachlev, thanks for raising this! I've opened a PR with a fix for this.

**Comment by bblommers** [comment]
Hi @jasonrdunne, can you share an example test that fails but shouldn't?

**Comment by bblommers** [comment]
Thanks @jasonrdunne - looks like it's mostly the last example that is failing? That should now be fixed as of `moto >= 5.0.11.dev17` - I'll try to get a proper release with this fix out later today.
